portfolio/src/main/webapp/script.js

```diff
@@ -23,21 +23,39 @@ function navtoggle() {
   document.getElementById('navmenu').classList.toggle('visible');
 }
 
-function openModal(target, index) {
+async function openModal(target, index) {
   const modal = document.getElementById('modal');
 
   modal.style.display = 'block';
   
-  const image = document.getElementById('focus-image');
-  const caption = document.getElementById('focus-image-caption');
-  const commSection = document.getElementById('comment-section');
+  const image = document.getElementById('focus-image'),
+        caption = document.getElementById('focus-image-caption'),
+        imageID = document.getElementById('image-id'),
+        commSection = document.getElementById('previous-comments');
 
   image.src = target.src;
   caption.innerHTML = target.alt;
+  imageID.value = index;
+
+  const params = new URLSearchParams();
+  params.append('id', index);
+
+  const response = await fetch('/data?' + index, {
+                                                   method: 'GET'
+  });
+  const comments = await response.text();
+
+  commSection.innerHTML = comments;
```

Comment:
This makes the website vulnerable to a form of [HTML injection attack](https://en.wikipedia.org/wiki/Code_injection).
For example, if I enter `<b>this is a BOLD comment</b>` into the Comment: field of the form and submit, the text renders as bold (screenshot attached). A malicious user can inject code in the comment field to run every time the page gets loaded. In general, extra precautions are taken into validating and sanitizing user input in the web. For the use case of user comments, one way to address this is to return the comment text uninterpreted and make sure it's escaped when added to the HTML of the page. For presenting text in HTML, you can take a look at [jQuery's text function]( https://api.jquery.com/text/#text-text) and the [createTextNode DOM API]( https://developer.mozilla.org/en-US/docs/Web/API/Document/createTextNode). Styling can be done in the javascript, HTML and CSS side of the website; a common approach is to have the REST methods (in this case, `GET data/?index`) return a JSON interpretation of the content and format it and embed it appropriately in the HTML of the page using javascript and CSS classes.
![YtKoCCELSJF](https://user-images.githubusercontent.com/29306214/88565431-e1990480-d034-11ea-8699-4ab095772075.png)